### DISCUSSION TITLE
How many requests can developer do to your Weather Forecast API?

### DISCUSSION THREAD

**Discussion by sentirpasseul** [open-meteo/open-meteo#115] [General]
Hey, hi! I made an application of weather forecast. So I use your Weather Forecast API, how many requests can i get from there?

**Comment by patrick-zippenfenig** [comment]
Hi @sentirpasseul,

thanks for asking. The [fair use policy](https://open-meteo.com/en/features) is 10.000 calls per day for non-commercial use.

There is not yet any automated blocking system, but maybe at some point I have to implement such a system to ensure that single individuals do not abuse the API. For now, the server utilisation is ok.

Best, Patrick

**Reply by Sharan123** [reply]
Hi @patrick-zippenfenig 
I am not sure if this number is still active. I am a PhD student at Temple and I've tried to use the API call to download historical weather data from 2016-2022, however both on my campus WIFI and now on my personal WIFI I was able to run 6 and 3 calls respectively before receiving a 429 Error (too many requests). I am unsure whether there is additional limitations on the type of the call or the number of variables selected? 

Any help would be much appreciated!

**Reply by patrick-zippenfenig** [reply]
@Sharan123 yes, the number of weather variables and time range effects how many calls you can perform. The basic formula is `call_weight = nDays / 14 * nVariables / 10`. An API call with data from 2016 until 2022 (7 years) would be weighted `7 * 365 / 14 = 182.5`.

In addition to the daily limit, there is also an hourly limit of 5000 calls as well as a minutely limit of 600. Those limits are in place to prevent abuse by fetching a lot of data in short period of time, effectively overloading the API.

For how many locations and variables do you need data? Can you just throttle your program to fetch data thorough a larger timespan?

**Reply by patrick-zippenfenig** [reply]
Sorry forgot to divide by 10. The call weight for 7 years of data is `(7 * 365 / 14) * (1 / 10) = 18.25`. Therefore roughly 500 locations per day until you reach the daily API limit

**Reply by Sharan123** [reply]
I got it - with approximately 20 variables and 2287 days it comes to a weight of about 3267 which perfectly matches the 3 API Calls. 
I need it for about 1000 locations - but I need it only once (not daily) since this is a part of a research to see if weather data can be useful for certain prediction tasks. Naturally, with publication, I plan to cite as described at the bottom of the open meteo Historical Weather page. 

Thanks for the clarification!

**Reply by patrick-zippenfenig** [reply]
Ah, yes. 20 weather variables over 7 years, increases the call weight quickly. To be fair it is also a lot of data ;-). I have just send you an API key via email so you can get all 1000 locations quickly.

**Reply by cevaboyz** [reply]
Hello @patrick-zippenfenig,  I'm doing a statistical project and I have a list of 6187 locations in Italy.

The time frame is 2020-01-01 to 2023-09-26

The parameter that i'm using are:

- weathercode
- average 2mt temperature
- rain sum
- snow sum

I would like to split my data in order to not abuse the daily limits, I don't have a hurry.

I just need this data one shot, if it's not possible to remain in the free limits/use a one time api  like you have provided @Sharan123 with
is there a way to just purchase the data for my need?


I'm using R

**Reply by patrick-zippenfenig** [reply]
Hi @cevaboyz, is this project for a university thesis or another non-profit organisation? You can write me an email (ideally from your university email address). Alternatively, you can also purchase the ["API Professional"](https://open-meteo.com/en/pricing) subscription for a month, download the required data and cancel at the end of the month.

**Reply by cevaboyz** [reply]
Hi @patrick-zippenfenig, thank you for your reply.

It's for a personal portfolio project so I think the correct way would be to purchase the Professional API.

In order to avoid purchasing a plan way over my need, I would like to have a confirm about how many calls do I need, given your formula

I have 1022 days of historical data that I need

My API calls are all with the same parameters:

`https://archive-api.open-meteo.com/v1/archive?latitude=45.6167&longitude=9.25&start_date=2020-09-15&end_date=2023-09-27&daily=weathercode,temperature_2m_mean,rain_sum,snowfall_sum&windspeed_unit=ms&timezone=Europe%2FBerlin&models=best_match`

I need to repeat this call for each of my 6168 locations. (with different lat&long obviously)

```
1022/14 * 4/10 = 29.2
29.2*6168=180105.6
```
I'm can easily stay within the limits of the API Professional (Up to 5 million API calls monthly, access to historical and climate data) ?

**Reply by patrick-zippenfenig** [reply]
A personal portfolio project sounds non-commercial to me. Ideally, you could even publish it as open-source :)

Yes, your calculation is correct and you will be able to download it easily with the API Professional Plan. Alternatively, you could just download over 2 days. Or just contact me via email to get an API key.

**Reply by YWilix** [reply]
Hello @patrick-zippenfenig  ,
I'm making a non-commercial app . and I'm using the marine API to do a 7 day forecast for 1 parameter only . 
according to the formula my call_weight per request is (7/14) * (1/10) = 0.05 . 
does that mean that one request is considered like 0.05 API call ? 

Also , does making a request for 2 locations at the same time double the call_weight ?

**Comment by norwest1998** [comment]
I am requesting data to put on my model yacht clubs web site, idea is to show what the weather conditions are going to be like on the weekend and then capture the actuals to pair with the results, create analytical data on where people perform better, etc.

I collect the weather data daily through the Open Meteo API's, Daily, Hourly and Current.

I do three calls per day through a google sheet trigger:
url: https://api.open-meteo.com/v1/forecast?latitude=-33.7312&longitude=150.9629&daily=weather_code,uv_index_max,precipitation_sum,wind_speed_10m_max,wind_speed_10m_mean,winddirection_10m_dominant,temperature_2m_min,temperature_2m_max&timezone=auto&start_date=2026-01-19&end_date=2026-01-26

url: https://api.open-meteo.com/v1/forecast?latitude=-33.7312&longitude=150.9629&current=temperature_2m,relative_humidity_2m,apparent_temperature,precipitation,weather_code,cloud_cover,wind_speed_10m,wind_direction_10m,uv_index&timezone=auto&start_date=2026-01-19&end_date=2026-01-19

url: https://api.open-meteo.com/v1/forecast?latitude=-33.7312&longitude=150.9629&hourly=temperature_2m,apparent_temperature,rain,wind_speed_10m,pressure_msl,wind_direction_10m,relative_humidity_2m,wet_bulb_temperature_2m,uv_index&timezone=auto&start_date=2026-01-19&end_date=2026-01-20


I receive the following error message for all three:
Error	{ [Exception: Request failed for https://api.open-meteo.com returned code 429. Truncated server response: {"error":true,"reason":"Daily API request limit exceeded. Please try again tomorrow."} (use muteHttpExceptions option to examine full response)] name: 'Exception' }

However, when I run the functions from the editor they all run okay.

Is there something I am missing that is causing the triggered call to fail?

**Reply by MathiasPetey** [reply]
I have the exact same issue.
If you found any solution, could you let me know?

**Reply by terraputix** [reply]
Hi, unfortunately running requests like this from Google Sheets triggers against the free API will most likely be limited by the concurrency limiter because someone else already consumed the quota for the shared IP address used by the Google Apps Script infrastructure. There was a similar problem with Cloudflare workers (https://github.com/open-meteo/open-meteo/issues/1726), which now has a workaround in place, but I don't think we'll do (or can do) something similar for Google Sheets. 

My recommendation would be to simply adjust your workflow/infrastructure to run from an IP address that is not shared with others.
